### PR TITLE
Using number of compute units to set  gridSize

### PR DESCRIPTION
**Pull request by qianfengz**
Fix to [Issue 266](https://github.com/ROCmSoftwarePlatform/composable_kernel/issues/266)

### CODE REVIEW

File: include/ck/host_utility/stream_utility.hpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            if(dword & 0x1)
+                count++;
+
+            dword = dword >> 1;
+        };
+
+        return (count);
+    };
+
+    hip_check_error(hipExtStreamGetCUMask(stream_config.stream_id_, MAX_MASK_DWORDS, &cuMask[0]));
```
Comment (rocking5566) [inline include/ck/host_utility/stream_utility.hpp:35]: &cuMask[0] => cuMask

**Review by rocking5566 — COMMENTED** [review]
Good solution

File: include/ck/host_utility/stream_utility.hpp (hunk 1 above)
Comment (aosewski) [reply include/ck/host_utility/stream_utility.hpp:35]: Why not just getting  the `multiProcessorCount` property from the device ? https://docs.amd.com/bundle/HIP-API-Guide-v5.4.1/page/a01151.html#add8d9d2ad52aece9fd1dbe25c18d9d57

Comment (qianfengz) [reply include/ck/host_utility/stream_utility.hpp:35]: > Why not just getting the `multiProcessorCount` property from the device ? https://docs.amd.com/bundle/HIP-API-Guide-v5.4.1/page/a01151.html#add8d9d2ad52aece9fd1dbe25c18d9d57

Cause the CUs usable by a stream can be masked by the `hipExtStreamCreateWithCUMask()`,  multiple streams can be created on the same device,  we should concern about the stream, rather the device.

**Review by rocking5566 — APPROVED** [review]

**Pull request merged**